### PR TITLE
Create ci.yml

### PR DESCRIPTION
**Pull request by sn0wbi**
add github action based pipeline to build and test the service with every pull request against main branch and when main branch get´s pushed

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,28 @@
+# This workflow will do a clean installation of node dependencies, cache/restore them, build the source code and run tests across different versions of node
+# For more information see: https://help.github.com/actions/language-and-framework-guides/using-nodejs-with-github-actions
+
+name: Node.js CI
+
+on:
+  push:
```
Comment (markbrockhoff) [inline .github/workflows/ci.yml:7]: Is it really a good idea to run the pipeline on every push?
As you can see it's currently running two times alone for this pr. I get what you want to archive here, but I think for now running it just on PRs should be sufficient :)

**Review by markbrockhoff — CHANGES_REQUESTED** [review]

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+  pull_request:
+    branches: [ master ]
```
Comment (dirien) [inline .github/workflows/ci.yml]: Please rename the master to main please!

I am surprised the repo was created with the non inklusive and offensive term master

**Review by dirien — CHANGES_REQUESTED** [review]
Left a comment regarding the branch name.

File: .github/workflows/ci.yml (hunk 1 above)
Comment (dirien) [reply .github/workflows/ci.yml:7]: I would let the push but only do it for the main branch:

```yaml
on:
  push:
    branches:
      - 'main'
    tags:
      - 'v*'
  pull_request:
```

So the build gets executed after the Merge, when there is new code on the `main` branch.

Comment (sn0wbi) [reply .github/workflows/ci.yml:7]: changed 👍

File: .github/workflows/ci.yml (hunk 2 above)
Comment (sn0wbi) [reply .github/workflows/ci.yml]: changed 👍

**Review by markbrockhoff — APPROVED** [review]

**Review by dirien — APPROVED** [review]
LGTM!

**Pull request merged**